cmd/capibmadm/cmd/powervs/key/create.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type keyCreateOptions struct {
+	key     string
+	keyName string
+}
+
+// CreateSSHKeyCommand - child command of 'key' to create an SSH key.
+func CreateSSHKeyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an SSH key in the PowerVS environment.",
+		Example: `
+# Create an SSH key.
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm powervs key create --name <key-name> --key "<ssh-key>" --service-instance-id <service-instance-id> --zone <zone>`,
+	}
+
+	var keyCreateOption keyCreateOptions
+	cmd.Flags().StringVar(&keyCreateOption.key, "key", "", "SSH RSA key string within a double quotation marks. For example, \"ssh-rsa AAA... \".")
+	cmd.Flags().StringVar(&keyCreateOption.keyName, "name", "", "The name of the SSH key.")
+	_ = cmd.MarkFlagRequired("key")
+	_ = cmd.MarkFlagRequired("key-name")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		return createSSHKey(cmd.Context(), keyCreateOption)
+	}
+	return cmd
+}
+
+func createSSHKey(ctx context.Context, keyCreateOption keyCreateOptions) error {
+	logger := log.Log
+	logger.Info("Creating SSH key...")
+
+	auth := iam.GetIAMAuth()
+	accountID, err := utils.GetAccountID(ctx, auth)
+	if err != nil {
+		return err
+	}
+	sess, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+	keyClient := instance.NewIBMPIKeyClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
+
+	sshBody := models.SSHKey{Name: &keyCreateOption.keyName, SSHKey: &keyCreateOption.key}
```

Comment:
do we need any validation for the validness of the key or `Create` call will take of that?!

Reply:
@mkumatag, I tried to pass an invalid SSH-key to `--key`, there is supporting logic on the server's end to validate the key. The following error was returned :
```
failed to Create PI Key with error [POST /pcloud/v1/tenants/{tenant_id}/sshkeys][400] pcloudTenantsSshkeysPostBadRequest
&{Code:0 Description:capiadm-sshkey sshkey is invalid; please enter a valid ssh RSA key: ssh: no key found Error:bad request Message:}
```
I'll add supporting changes to validate the key before the request is made.

Reply:
Do we really need? or we just rely on server to take care of validation?!

Reply:
We can rely on server too. :) 
But I was only thinking if it would be better to validate the key locally as this overhead will be borne by the server. Your thoughts?